tuf_conformance/test_file_download.py

```diff
@@ -4,15 +4,19 @@
 from tuf_conformance.simulator_server import SimulatorServer
 from tuf_conformance.client_runner import ClientRunner
 from tuf_conformance import utils
+from tuf_conformance.utils import TestTarget
 
 from tuf.api.metadata import (
     Targets, Snapshot
 )
 
-class TestTarget:
-    path: str
-    content: bytes
-    encoded_path: str
+def get_url_prefix(server_process_handler: utils.TestServerProcess,
+                   client: ClientRunner) -> str:
+    url_prefix = (
+        f"http://{utils.TEST_HOST_ADDRESS}:"
+        f"{server_process_handler.port!s}/{os.path.basename(client._remote_target_dir.name)}"
```

Comment:
is the path correct here? the SimulatorServer uses the name of the testcase (or something like that) as the repos key, not the basename of the tmpdir

Reply:
cc @AdamKorcz

Reply:
@segiddins I am looking into this. Have you observed unexpected behavior as a precursor for asking?

Reply:
```
Failed to download target #<Sigstore::TUF::Targets::Target:0x000000011c467838 @path="target_file.txt", @length=15, @hashes={"sha256"=>"5243e1438ef82951413c58d9cf1f2a1d983588f6c3530c3b661bda6c99629a61"}> "5243e1438ef82951413c58d9cf1f2a1d983588f6c3530c3b661bda6c99629a61.target_file.txt" from http://127.0.0.1:50415/tmpq063zyq5/5243e1438ef82951413c58d9cf1f2a1d983588f6c3530c3b661bda6c99629a61.target_file.txt: 404 "Did not find repository for tmpq063zyq5"
```

Reply:
from:

```ruby
["sigstore_tuf_download",
 "--metadata-url",
 "http://127.0.0.1:50414/test_downloaded_file_is_correct2/metadata/",
 "--metadata-dir",
 "/var/folders/p3/950_j_ps7xq47vrzdj1qvg940000gn/T/tmpy0x6fz6i/metadata",
 "target_file.txt",
 "--targets-dir",
 "/var/folders/p3/950_j_ps7xq47vrzdj1qvg940000gn/T/tmpgh88t5zi",
 "--target-base-url",
 "http://127.0.0.1:50415/tmpq063zyq5"]
 ```

Reply:
Which test is failing?

Reply:
I _think_ the problem is that the test does not prefix the target file with the files hash. Your call shows that you are downloading `http://127.0.0.1:50415/tmpq063zyq5/5243e1438ef82951413c58d9cf1f2a1d983588f6c3530c3b661bda6c99629a61.target_file.txt`, but I _think_ it should be available at `http://127.0.0.1:50415/tmpq063zyq5/target_file.txt`.

Reply:
all the `tuf_conformance/test_file_download.py` tests continue to fail in the same way when i change the default config to `prefix_targets_with_hash: false`

Reply:
PEBKAC... My http client was accidentally making requests against the metadata URL since ruby's Net::HTTP behavior is... confusing